bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/views/ReminderView.java

```diff
@@ -902,4 +914,39 @@ public void refresh() {
 			cv.notify(CommonViewer.Message.update);
 		}, cv);
 	}
-}
+
+	private void initializeSortByDueDateAction(final ReminderViewCommonContentProvider contentProvider) {
+		System.out.println("state  33 " + state2);
+		if (state2 == 0) {
+			contentProvider.setComparator(new Comparator<Reminder>() {
+				@Override
+				public int compare(Reminder o1, Reminder o2) {
+					ConfigServiceHolder.setUser(Preferences.USR_SORT_BY_DUE_DATE, 0);
+					return TimeTool.compare(o2.getDateDue(), o1.getDateDue());
+
+				}
+			});
+			state = 1;
+			sortByDueDate.setText(Messages.ReminderView_sortByDueDateAscending);
+			sortByDueDate.setChecked(true);
+		} else if (state2 == 1) {
+			
+			contentProvider.setComparator(new Comparator<Reminder>() {
+				@Override
+				public int compare(Reminder o1, Reminder o2) {
+					ConfigServiceHolder.setUser(Preferences.USR_SORT_BY_DUE_DATE, 1);
+					return TimeTool.compare(o1.getDateDue(), o2.getDateDue());
+				}
+			});
+			state = 2;
+			sortByDueDate.setText(Messages.ReminderView_sortByDueDateDescending);
+			sortByDueDate.setChecked(true);
+		} else if (state2 == 2) {
+			ConfigServiceHolder.setUser(Preferences.USR_SORT_BY_DUE_DATE, 2);
+			contentProvider.setComparator(null);
+			state = 0;
+			sortByDueDate.setText(Messages.ReminderView_sortByDueDate);
+			sortByDueDate.setChecked(false);
+		}
+	}
```

Comment:
Weshalb hier nochmals des selbe code wie in der Action ab Zeile 459/469?

Das sollte nicht nötig sein, wenn in der Action auch der Status mit dem ConfigService gespeichert wird.